blocks/init/src/blocks/components/drawer/assets/drawer.js

```diff
@@ -0,0 +1,59 @@
+export class Drawer {
+  constructor(
+    selector = '.js-drawer',
+    CLASS_IS_OPEN = 'menu-is-open',
+    CLASS_OVERLAY_IS_SHOWING = 'page-overlay-shown',
+    CLASS_NO_SCROLL = 'u-no-scroll'
+  ) {
+    this.CLASS_IS_OPEN = CLASS_IS_OPEN;
+    this.CLASS_OVERLAY_IS_SHOWING = CLASS_OVERLAY_IS_SHOWING;
+    this.CLASS_NO_SCROLL = CLASS_NO_SCROLL;
+    this.drawer = document.querySelector(selector);
+    this.trigger = document.querySelector(`.${this.drawer.getAttribute('data-trigger')}`);
+    this.overlay = document.querySelector(`.${this.drawer.getAttribute('data-overlay')}`);
+  }
+  
+  preventScroll() {
+    document.body.style.top = `-${window.scrollY}px`;
+    document.body.classList.add(this.CLASS_NO_SCROLL);
+  }
+  
+  enableScroll() {
+    const scrollY = document.body.style.top;
+    document.body.classList.remove(this.CLASS_NO_SCROLL);
+    document.body.style.top = '';
+    window.scrollTo(0, parseInt(scrollY || '0', 10) * -1);
+  }
+  
+  openMobileMenu() {
+    document.body.classList.add(this.CLASS_IS_OPEN);
+    document.body.classList.add(this.CLASS_OVERLAY_IS_SHOWING);
+    this.preventScroll();
+  }
+  
+  closeMobileMenu() {
+    document.body.classList.remove(this.CLASS_IS_OPEN);
+    document.body.classList.remove(this.CLASS_OVERLAY_IS_SHOWING);
+    this.enableScroll();
+  }
+  
+  closeMobileMenuOnOverlayClick() {
+    if (this.overlay) {
+      this.overlay.addEventListener('click', () => {
+        this.closeMobileMenu();
+      });
+    }
+  }
+
+  drawerInit() {
+    this.trigger.addEventListener('click', () => {
+      if (document.body.classList.contains(this.CLASS_IS_OPEN)) {
+        this.closeMobileMenu();
+      } else {
+        this.openMobileMenu();
+      }
```

Comment:
```suggestion
      if (document.body.classList.contains(this.CLASS_IS_OPEN)) {
        this.closeMobileMenu();
      } 
      
      this.openMobileMenu();
```

I think it should be safe to remove `else` part.